### DISCUSSION TITLE
Decryption of shadow fight 2 and shades files.

### DISCUSSION THREAD

**Discussion by VladiSlav0023** [community/community#75010] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello, I have a question, have you encountered encrypted files that serve as decryption? If you don’t have these, then I don’t know how to decrypt 3 files. 2 of them practically do the same encoding (like AES), but 3 encryptions can be decrypted through the lib key in the apk through the loading screen, which it would create a decrypted file or decrypt inside the file itself in the video mdl_data.bin (that’s what I see in the video showed) ) you just need to know what needs to be changed, since there are more than 10 files, so,
[perks_xml.txt](https://github.com/community/community/files/13317889/perks_xml.txt)
[mdl_helm_camera_xml.txt](https://github.com/community/community/files/13317890/mdl_helm_camera_xml.txt)
[mdl_armor_light_act1_common_bin_xml.txt](https://github.com/community/community/files/13317891/mdl_armor_light_act1_common_bin_xml.txt)

**Comment by mireaunofficial** [comment, marked as answer]
Тебе ещё нужна помощь с этим?

**Reply by VladiSlav0023** [reply]
да, а ты знаешь как расшифровать ?

**Reply by mireaunofficial** [reply]
Вот что я смог сделать
https://t.me/s/BotBotOfficial/47
Мне потребовалось пару недель в Ghidra чтобы найти

**Reply by VladiSlav0023** [reply]
этот скрипт который ты расшифровал moves может расшифровать и другие файлы по типу shades и модели ?

**Reply by WorldLyBlackSmith** [reply]
> Тебе ещё нужна помощь с этим?

Можете ли вы прислать мне файл? Как я могу расшифровать его самостоятельно?

**Reply by VladiSlav0023** [reply]
> > Тебе ещё нужна помощь с этим?
> 
> Можете ли вы прислать мне файл? Как я могу расшифровать его самостоятельно?

А то что я скидывал этого не достаточно ? Расшифровать можно лишь если знаешь какой метод шифрования там используется

**Comment by Cobb787** [comment]
Interesting technical question about decrypting Shadow Fight 2 and Shades files, especially the different encryption methods and the role of the APK libraries during loading. Understanding how these files are processed could help clarify whether the game decrypts them internally or uses a separate mechanism. Hopefully someone experienced with the game’s file structure can share useful insights about the encryption formats and relevant tools or resources. For anyone interested in more Shadow Fight 2-related information, I’d also recommend checking out [Shadow Fight 2](https://shadowfight2apkgame.com), which offers useful game content and resources for fans.
